smoketest.sh

```diff
@@ -10,16 +10,54 @@ function runCryostat() {
     local grafanaPort="$(xpath -q -e 'project/properties/cryostat.itest.grafana.port/text()' pom.xml)"
     # credentials `user:pass`
     echo "user:d74ff0ee8da3b9806b18c877dbf29bbde50b5bd8e4dad7a3a725000feb82e8f1" > "./conf/cryostat-users.properties"
+
+    if [ "$1" = "postgres" ]; then
+        JDBC_URL="jdbc:postgresql://cryostat:5432/cryostat"
+        JDBC_DRIVER="org.postgresql.Driver"
+        HIBERNATE_DIALECT="org.hibernate.dialect.PostgreSQL95Dialect"
+        JDBC_USERNAME="postgres"
+        JDBC_PASSWORD="abcd1234"
+        HBM2DDL="update"
+    elif [ "$1" = "h2file" ]; then
+        JDBC_URL="jdbc:h2:file:/opt/cryostat.d/conf.d/h2;INIT=create domain if not exists jsonb as other"
+        HBM2DDL="update"
+    fi
+
     GRAFANA_DATASOURCE_URL="http://${host}:${datasourcePort}" \
         GRAFANA_DASHBOARD_URL="http://${host}:${grafanaPort}" \
         CRYOSTAT_RJMX_USER=smoketest \
         CRYOSTAT_RJMX_PASS=smoketest \
         CRYOSTAT_ALLOW_UNTRUSTED_SSL=true \
         CRYOSTAT_REPORT_GENERATOR="http://${host}:10001" \
         CRYOSTAT_AUTH_MANAGER=io.cryostat.net.BasicAuthManager \
+        CRYOSTAT_JDBC_URL="$JDBC_URL" \
+        CRYOSTAT_JDBC_DRIVER="$JDBC_DRIVER" \
+        CRYOSTAT_HIBERNATE_DIALECT="$HIBERNATE_DIALECT" \
+        CRYOSTAT_JDBC_USERNAME="$JDBC_USERNAME" \
+        CRYOSTAT_JDBC_PASSWORD="$JDBC_PASSWORD" \
+        CRYOSTAT_HBM2DDL="$HBM2DDL" \
         exec "$DIR/run.sh"
 }
 
+function runPostgres() {
+    if [ ! -d "$(dirname $0)/conf/postgres" ]; then
+        mkdir "$(dirname $0)/conf/postgres"
+    fi
+    local image="$(xpath -q -e 'project/properties/postgres.image/text()' pom.xml)"
+    local version="$(xpath -q -e 'project/properties/postgres.version/text()' pom.xml)"
+    podman run \
+        --name postgres \
+        --pod cryostat-pod \
+        --env POSTGRES_USER=postgres \
+        --env POSTGRES_PASSWORD=abcd1234 \
+        --mount type=bind,source="$(dirname $0)/conf/postgres",destination=/var/lib/postgresql/data/pgdata,relabel=shared \
+        --mount type=bind,source="$(dirname $0)/src/test/resources/postgres",destination=/docker-entrypoint.initdb.d,relabel=shared \
+        --env PGDATA=/var/lib/postgresql/data/pgdata \
+        --rm -d "${image}:${version}"
+    sleep 5
```

Comment:
Is the sleep there because it gives time for the pod to start running the postgres instance before connecting?

Reply:
Exactly, there are probably smarter and more reliable ways to do this but it seemed to work well enough for this script, for now.